api/provider/keycloak.go

```diff
@@ -0,0 +1,94 @@
+package provider
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	"github.com/netlify/gotrue/conf"
+	"golang.org/x/oauth2"
+)
+
+// Keycloak
+type keycloakProvider struct {
+	*oauth2.Config
+	Host string
+}
+
+type keycloakUser struct {
+	Name          string `json:"name"`
+	Sub           string `json:"sub"`
+	Email         string `json:"email"`
+	EmailVerified bool   `json:"email_verified"`
+}
+
+// NewKeycloakProvider creates a Keycloak account provider.
+func NewKeycloakProvider(ext conf.OAuthProviderConfiguration, scopes string) (OAuthProvider, error) {
+	if err := ext.Validate(); err != nil {
+		return nil, err
+	}
+
+	oauthScopes := []string{
+		"profile",
+		"email",
+	}
+
+	if scopes != "" {
+		oauthScopes = append(oauthScopes, strings.Split(scopes, ",")...)
+	}
+
+	if ext.URL == "" {
+		return nil, errors.New("Unable to find URL for the Keycloak provider")
+	}
+	host := chooseHost(ext.URL, "")
```

Comment:
is there a default url for keycloak? if not we might wanna do additional validation here - if `ext.URL` is empty, return an error